lib/rails/generators/channel/channel_generator.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails/generators/named_base'
+
+module Rails # :nodoc:
+  module Generators # :nodoc:
+    class ChannelGenerator < Rails::Generators::NamedBase # :nodoc:
+      desc 'This generator creates a cable channel file at app/channels'
+      source_root File.expand_path('../templates', __FILE__)
+      check_class_collision suffix: 'Channel'
+
+      class_option :parent, :type => :string, :desc => 'The parent class for the generated channel'
+
+      hook_for :test_framework
+
+      def create_channel_files
+        template 'channel.rb.erb', File.join('app/channels', class_path, "#{file_name}_channel.rb")
+        template 'channel.coffee.erb', File.join('app/assets/javascripts/channels', "#{file_name}_channel.coffee")
+      end
+
+      def channel_name
+        class_name.underscore
+      end
+
+      def channel_class
+        "#{class_name}Channel"
+      end
+
+      def parent_class_name
```

Comment:
Don't think we need this. It should just always be ApplicationCable::Channel. Just like the controller generator always generates controllers inheriting from ApplicationController.